lib/ui/painting/image_encoding.cc

```diff
@@ -181,17 +206,29 @@ sk_sp<SkData> EncodeImage(sk_sp<SkImage> p_image,
 }
 
 void EncodeImageAndInvokeDataCallback(
-    std::unique_ptr<DartPersistentValue> callback,
     sk_sp<SkImage> image,
-    GrContext* context,
+    std::unique_ptr<DartPersistentValue> callback,
+    ImageByteFormat format,
     fml::RefPtr<fml::TaskRunner> ui_task_runner,
-    ImageByteFormat format) {
-  sk_sp<SkData> encoded = EncodeImage(std::move(image), context, format);
-
-  ui_task_runner->PostTask(
-      fml::MakeCopyable([callback = std::move(callback), encoded]() mutable {
+    fml::RefPtr<fml::TaskRunner> gpu_task_runner,
+    fml::RefPtr<fml::TaskRunner> io_task_runner,
+    GrContext* resource_context,
+    fml::WeakPtr<SnapshotDelegate> snapshot_delegate) {
+  auto callback_task = fml::MakeCopyable(
+      [callback = std::move(callback)](sk_sp<SkData> encoded) mutable {
         InvokeDataCallback(std::move(callback), std::move(encoded));
-      }));
+      });
+
+  auto encode_task = [callback_task = std::move(callback_task), format,
+                      ui_task_runner](sk_sp<SkImage> raster_image) {
+    sk_sp<SkData> encoded = EncodeImage(std::move(raster_image), format);
+    ui_task_runner->PostTask(
```

Comment:
Minor optimization: If you use `RunNowOrPostTask`, you will avoid posting to the UI task runner in the case where the raster image acquisition could not be done on the GPU task runner and you had to post to the IO for the same. This works well too though and I understand if you'd rather keep it this way for clarity.